src/main/kotlin/no/nav/familie/ba/sak/integrasjoner/økonomi/utbetalingsoppdrag/BehandlingsinformasjonUtleder.kt

```diff
@@ -0,0 +1,86 @@
+import no.nav.familie.ba.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ba.sak.kjerne.fagsak.Fagsak
+import no.nav.familie.ba.sak.kjerne.fagsak.FagsakType
+import no.nav.familie.ba.sak.kjerne.vedtak.Vedtak
+import no.nav.familie.felles.utbetalingsgenerator.domain.Behandlingsinformasjon
+import no.nav.familie.felles.utbetalingsgenerator.domain.IdentOgType
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.time.YearMonth
+
+@Component
+class BehandlingsinformasjonUtleder(
+    private val endretMigreringsdatoUtleder: EndretMigreringsdatoUtleder,
+    private val clockProvider: ClockProvider,
+) {
+    fun utled(
+        saksbehandlerId: String,
+        vedtak: Vedtak,
+        forrigeTilkjentYtelse: TilkjentYtelse?,
+        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
+        erSimulering: Boolean,
+    ): Behandlingsinformasjon {
+        val behandling = vedtak.behandling
+        val fagsak = behandling.fagsak
+        val endretMigreringsdato = endretMigreringsdatoUtleder.utled(fagsak, forrigeTilkjentYtelse)
+        return Behandlingsinformasjon(
+            saksbehandlerId = saksbehandlerId,
+            behandlingId = behandling.id.toString(),
+            eksternBehandlingId = behandling.id,
+            eksternFagsakId = fagsak.id,
+            fagsystem = FagsystemBA.BARNETRYGD,
+            personIdent = fagsak.aktør.aktivFødselsnummer(),
+            vedtaksdato = vedtak.vedtaksdato?.toLocalDate() ?: LocalDate.now(clockProvider.get()),
+            opphørAlleKjederFra = finnOpphørsdatoForAlleKjeder(forrigeTilkjentYtelse, sisteAndelPerKjede, endretMigreringsdato),
+            utbetalesTil = finnUtebetalesTil(fagsak),
+            opphørKjederFraFørsteUtbetaling = finnOpphørKjederFraFørsteUtbetaling(endretMigreringsdato, erSimulering),
+        )
+    }
+
+    private fun finnOpphørsdatoForAlleKjeder(
+        forrigeTilkjentYtelse: TilkjentYtelse?,
+        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
+        endretMigreringsdato: YearMonth?,
+    ): YearMonth? {
+        if (forrigeTilkjentYtelse == null || sisteAndelPerKjede.isEmpty()) {
+            return null
+        }
+        if (endretMigreringsdato != null) {
+            return endretMigreringsdato
+        }
+        return null
+    }
+
+    private fun finnUtebetalesTil(fagsak: Fagsak): String =
+        when (fagsak.type) {
+            FagsakType.NORMAL,
+            FagsakType.BARN_ENSLIG_MINDREÅRIG,
+            -> fagsak.aktør.aktivFødselsnummer()
+
+            FagsakType.INSTITUSJON,
+            -> {
+                val tssEksternId = fagsak.institusjon?.tssEksternId
+                if (tssEksternId == null) {
+                    throw IllegalStateException(
+                        "Fagsak ${fagsak.id} er av type institusjon og mangler informasjon om institusjonen",
+                    )
+                }
```

Comment:
```suggestion
                              val tssEksternId = fagsak.institusjon?.tssEksternId ?: throw IllegalStateException(
                    "Fagsak ${fagsak.id} er av type institusjon og mangler informasjon om institusjonen",
                )
                
                tssEksternId
```

Litt smak og behag her også, men vi har elvis operator i kotlin som kan brukes.